queries/fetch_backruns.sql

```diff
@@ -1,37 +1,49 @@
-SELECT 
-    t.hash,
-    t.from,
-    t.to,
-    t.gas,
-    t.gas_price,
-    t.value,
-    t.input AS calldata,
-    b.number AS block_number,
-    b.timestamp AS block_timestamp,
-    r.cumulative_gas_used,
-    r.status,
-    tr.trace_type,
-    tr.action_from AS internal_from,
-    tr.action_to AS internal_to,
-    tr.action_value AS internal_value
-FROM ethereum.transactions t
-JOIN ethereum.blocks b ON t.block_number = b.number
-JOIN ethereum.receipts r ON t.hash = r.transaction_hash
-LEFT JOIN ethereum.traces tr ON t.hash = tr.transaction_hash
-WHERE b.number BETWEEN :start_block AND :end_block
--- This query extracts detailed transaction information from Ethereum blockchain.
--- It joins transaction, block, receipt, and trace data to provide a comprehensive view.
--- Parameters :start_block and :end_block define the range of blocks to be analyzed.
--- Trace information provides insight into internal contract interactions.
+WITH 
+latest_transactions AS (
+    SELECT 
+        from_hex(CAST(json_extract(transactions, '$[0].hash') AS VARCHAR)) AS tx_hash,
+        transactions,
+        blockNumber,
+        to_unixtime(cast(from_unixtime(timestamp / 1000) as timestamp)) as unix_timestamp
+    FROM mevblocker.raw_bundles
+    WHERE to_unixtime(cast(from_unixtime(timestamp / 1000) as timestamp)) > to_unixtime(NOW() - INTERVAL '1' HOUR)
+),
 
+bundles_with_backruns AS (
```

Comment:
This also fetches bundles without backruns (which is fine, but the name is confusing).

Overall I'm not sure why you need three sub queries here, I'd just go with a single:

```sql
SELECT block_number, timestamp, transactions
FROM mevblocker.raw_bundles t
JOIN ethereum.transactions e
  ON t.blockNumber = e.block_number
  AND from_hex(CAST(json_extract(transactions, '$[0].hash') AS VARCHAR)) = e.hash
  AND block_number >= {{start_block}}
  AND block_number <= {{end_block}}
```

This will fetch all the submitted backruns for all mevblocker transactions on the block they were ultimately included.

Reply:
Are we good with this?:
`WITH parsed_transactions AS (
    SELECT
        t.blockNumber AS block_number,
        t.timestamp,
        t.transactions,
        CAST(json_extract(t.transactions, '$[0].hash') AS VARCHAR) AS transaction_hash
    FROM
        mevblocker.raw_bundles t
)
SELECT 
    pt.block_number, 
    pt.timestamp, 
    pt.transactions,
    e.hash AS transaction_hash,
    e."from",
    e.to,
    e.gas_limit,
    e.gas_price,
    e.data AS calldata,
    e.gas_used
FROM parsed_transactions pt
JOIN ethereum.transactions e
  ON pt.block_number = e.block_number
  AND from_hex(pt.transaction_hash) = e.hash
WHERE pt.block_number >= {{start_block}}
  AND pt.block_number <= {{end_block}}
ORDER BY pt.timestamp DESC;`

Only thing is that is kinda hacky, as we must calculate blocks using python (then pass it to dune), and as the script runs periodically (e.g 1 hour or whatever), the amount of blocks we're searching in past for the same amount of time can be different.
Just want to ensure the edge cases and that all meets the requirements.

Reply:
Why do you care about all the fields from `ethereum.transactions`? How is this going to help you in the later parts of the task? 
I think you still need to fetch each block's content from the full node (because you want all the other transactions that were part of the block as well). You could do that in Dune as well, but in any case I would separate the two queries as they are conceptually different things.

Regarding block numbers vs. timestamps I don't have a strong feeling. The goal for this pipeline is to ultimately run for all blocks, so I'd assume it to have some persistence of what was the last processed block and then be able to re-run from there. But of course this is also possible with timestamps.

Reply:
Hm, I think we got misunderstanding here.
I previously were including all possible info to gather whatever it is needed to include all the backruns.
As checked now, your implementation seems better in this point.
In this case, If everything is okay, I will make a new commit with leftover python implementation.
Before that, Do you want to include the query itself on a specific account, or for rn I'll just upload it to mine?

Thanks!

Reply:
You can upload it to yours, as long as it's also checked in, whatever account is running this repo later can re-create the query and specify the query id in the configuraiton.

Reply:
Hello!
Made a commit with everything mentioned above to master branch

Reply:
Why not to this PR branch? This allows me to review just the changes we discussed here and comment on them (now I need to find the relevant changes in a commit titled "A lot of QOL changes and leftover python implementation" which doesn't seem standard engineering practice)

Reply:
My fault. Messed up with test branch difference.
Anyway, I will reset the branch now and create PR (so It would be easier to you:)